src/util.jl

```diff
@@ -39,22 +39,22 @@ end
 @pure tuple_minimum(T::Tuple) = minimum(T)
 
 # Something doesn't match up type wise
-function check_array_parameters(Size, T, N, L)
-    (!isa(Size, DataType) || (Size.name !== Tuple.name)) && throw(ArgumentError("Static Array parameter Size must be a Tuple type, got $Size"))
-    !isa(T, Type) && throw(ArgumentError("Static Array parameter T must be a type, got $T"))
-    !isa(N.parameters[1], Int) && throw(ArgumenError("Static Array parameter N must be an integer, got $(N.parameters[1])"))
-    !isa(L.parameters[1], Int) && throw(ArgumentError("Static Array parameter L must be an integer, got $(L.parameters[1])"))
+@generated function check_array_parameters(Size, T, N, L)
```

Comment:
Oh.... That probably wasn't intended... I got a bit confused because I started testing with an old version of master, where this was still a generated function...